js/data.js

```diff
@@ -6,3 +6,13 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+addEventListener('beforeunload', function (event) {
```

Comment:
`addEventListener` is a method that needs to be attached to an object. You should be calling this on the `window` object